src/library/OverflowContainer/OverflowContainerWithShadows.js

```diff
@@ -49,34 +56,72 @@ const ThemedOverflowContainer = createThemedComponent(
     )
 );
 
-const Root = createStyledComponent(
-  ThemedOverflowContainer,
-  ({ hasShadowBottom, hasShadowTop, theme: baseTheme }) => {
-    const theme = componentTheme(baseTheme);
-
-    const boxShadows = [];
-    hasShadowTop &&
-      boxShadows.push(theme.OverflowContainerWithShadows_boxShadowTop);
-    hasShadowBottom &&
-      boxShadows.push(theme.OverflowContainerWithShadows_boxShadowBottom);
-    const boxShadow = boxShadows.length ? boxShadows.join(',') : undefined;
+const getBoxShadows = ({ theme: baseTheme, ...restProps }) => {
+  const theme = componentTheme(baseTheme);
+  const boxShadows = [];
+  Object.keys(restProps)
+    .filter((prop) => prop.indexOf('hasShadow') !== -1)
+    .forEach((prop) => {
+      if (restProps[prop]) {
+        const edge = prop.split('hasShadow')[1];
+        boxShadows.push(theme[`OverflowContainerWithShadows_boxShadow${edge}`]);
+      }
+    });
+  return boxShadows.length ? boxShadows.join(',') : undefined;
+};
 
-    // [1] - 1px to avoid unwanted vertical scrollbar,
+const styles = {
+  scroller: ({ scrollX, scrollY }) => {
+    // [1] - 1px to avoid unwanted scrollbar,
     //       1px in to avoid potentially cutting off of focus ring of
     //       subcomponents in body
-    return {
-      boxShadow,
-      overflowX: 'hidden',
-
-      paddingBottom: 2, // [1]
-      paddingTop: 2 // [1]
-    };
+    let styles = [
+      {
+        flex: '1 1 auto'
+      }
+    ];
+    if (scrollX) {
```

Comment:
Rather than defining styles and later conditionally mutating it, can we follow this approach instead which avoids the mutation?  (This comment likely applies elsewhere too.)  

```js
const styles = [
  {
    flex: '1 1 auto',

    ...(scrollX
      ? {
          overflowY: 'hidden',
          paddingLeft: 2, // [1]
          paddingRight: 2 // [1]
        }
      : undefined)
  }
];
```

Reply:
The scroller style function could actually then just be converted to use an implicit return with no need to define `styles` at all.